src/page-multi-facility/ReferenceDataModal/SyncNewReferenceData.tsx

```diff
@@ -0,0 +1,105 @@
+import { isEmpty } from "lodash";
+import React, { useState } from "react";
+import styled from "styled-components";
+
+import { referenceFacilitiesProp } from "../../database";
+import { useFacilities } from "../../facilities-context";
+import useScenario from "../../scenario-context/useScenario";
+import { Facility, ModelInputs, ReferenceFacility } from "../types";
+import ReferenceDataModal from ".";
+import {
+  getUnmappedFacilities,
+  getUnmappedReferenceFacilities,
+  ReferenceFacilitySelect,
+  ReferenceFacilitySelections,
+  TitleContainer,
+  TitleText,
+} from "./shared";
+
+const Spacer = styled.span`
+  margin-right: 2em;
+`;
+
+const Title: React.FC<Pick<Props, "stateName" | "systemType">> = ({
+  stateName,
+  systemType,
+}) => (
+  <TitleContainer>
+    <TitleText>Prepopulate Data</TitleText>
+    <TitleText>
+      We've found new facility data - select a corresponding facility to
+      autofill with case data
+    </TitleText>
+    <TitleText>
+      State: {stateName}
+      <Spacer />
+      Type of System: {systemType}
+    </TitleText>
+  </TitleContainer>
+);
+
+interface Props {
+  open: boolean;
+  stateName: ModelInputs["stateName"];
+  systemType: Facility["systemType"];
+  onClose: () => void;
+}
+
+const SyncNewReferenceData: React.FC<Props> = ({
+  open,
+  stateName,
+  systemType,
+  onClose,
+}) => {
+  const [selections, setSelections] = useState<ReferenceFacilitySelections>({});
+  const [scenarioState] = useScenario();
+  const {
+    state: { facilities: facilitiesMapping, referenceFacilities },
+  } = useFacilities();
+  const scenario = scenarioState.data;
+  const mappedReferenceFacilities = scenario?.[referenceFacilitiesProp] || {};
+  const unmappedFacilities = getUnmappedFacilities(
+    mappedReferenceFacilities,
+    facilitiesMapping,
+  );
+  const unmappedReferenceFacilities = getUnmappedReferenceFacilities(
+    mappedReferenceFacilities,
+    referenceFacilities,
+  );
+
+  if (!open || isEmpty(unmappedReferenceFacilities)) return null;
```

Comment:
Ok, cool.  So I think after this line, we'll be left with `unmappedReferenceFacilities` that we can either:

1.  Map to existing users facilities if they aren't already mapped

or

2.  Add as an entirely new facility

Reply:
Yeah, should I add a new issue to add that new functionality after this merges? Currently they won't yet be able to add as entirely new facility.

Reply:
Just got off the phone with Daniela.  We're going to leave this as-is for now so that we can merge since it looks like the issue Sokunthel is working on (#556) may be implementing this functionality.